movai_core_shared/core/zmq_client.py

```diff
@@ -129,69 +116,51 @@ def _extract_reponse(self, buffer: bytes):
             )
             return {}
 
-    def recieve(self) -> dict:
-        """
-        Recieves a message response over ZeroMQ from the server.
-
-        Raises:
-            MessageError: In case response is empty.
-
-        Returns:
-            dict: The response from the server.
-        """
-        buffer = self._recieve()
-        response = self._extract_reponse(buffer)
-        return response
-
 
 class AsyncZMQClient(ZMQClient):
     """An Async implementation of ZMQ Client"""
 
-    def _init_context(self):
-        self._zmq_ctx = zmq.asyncio.Context()
-
-    def _init_lock(self):
-        self._lock = asyncio.Lock()
-
-    async def _send(self, msg: bytes):
-        """Asynchrounously send the message.
+    def __init__(self, identity: str, server_addr: str) -> None:
+        """Initializes the object and the connection to the server.
 
         Args:
-            data (bytes): the msg representation
+            identity (str): A unique identity which will be used by
+            server_addr (str): The server addr and port
         """
-        async with self._lock:
-            await self._socket.send(msg)
+        super().__init__(identity, server_addr)
+        self._lock = asyncio.Lock()
+        self._init_context()
+        self._socket = self.prepare_socket()
+
+    def _init_context(self):
+        self._zmq_ctx = zmq.asyncio.Context()
 
-    async def send(self, msg: dict) -> None:
+    async def send(self, msg: dict, response_required: bool = False) -> dict:
         """
         Send the message request over ZeroMQ to the local robot message server.
 
         Args:
             msg (dict): The message request to be sent
+            response_required (bool): Will wait for response, Default False
         """
         data = self._create_msg(msg)
-        await self._send(data)
-
-    async def _recieve(self):
-        """Asynchrounsly recieves data from the server.
-
-        Returns:
-            (bytes): raw data from the server.
-        """
         async with self._lock:
-            buffer = await self._socket.recv_multipart()
-            return buffer
+            await self._socket.send(data)
+        if response_required:
+            response = await self._receive()
+        else:
+            response = {}
+        return response
 
-    async def recieve(self) -> dict:
+    async def _receive(self) -> dict:
```

Comment:
where did "recieve" go ?
also breaks the api

Reply:
no, it's not 
no need in receive function 
because now you can send and get response from the same calling
and again nothing is using the zmq_client directly 
only through the interface of the message_client